### PR TITLE
[DEV-5532] Define dashboard related DB models

### PR DESCRIPTION
**Pull request by gaetano-guerriero**
This change is [Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3130)

### CODE REVIEW

File: server/athenian/api/models/state/models.py [hunk 1]
```diff
@@ -484,3 +484,68 @@ class Share(create_time_mixin(created_at=True), Base):
...
+
+    __tablename__ = "team_dashboards"
+
+    id = Column(Integer(), primary_key=True)
+    team_id = Column(
+        Integer(),
+        ForeignKey("teams.id", name="fk_team_dashboard_team", ondelete="CASCADE"),
+        nullable=False,
+    )
+    """The team this dashboard belongs to."""
```
Comment (vmarkovtsev) [inline server/athenian/api/models/state/models.py:500]: `Column(comment="The team this dashboard belongs to.")`

Comment (vmarkovtsev) [reply server/athenian/api/models/state/models.py:500]: Please update all the encounters

File: server/athenian/api/models/state/models.py [hunk 2]
```diff
@@ -484,3 +484,68 @@ class Share(create_time_mixin(created_at=True), Base):
...
+    """Position of the chart in the containing dashboard."""
+
+    metric = Column(String, nullable=False)
+    """The metric this chart is displaying."""
+
+    time_from = Column(TIMESTAMP(timezone=True), nullable=True)
+    time_to = Column(TIMESTAMP(timezone=True), nullable=True)
+    """The fixed time interval of the chart.
+
+    If `time_from` and `time_to` are expressed `time_interval` will be NULL.
```
Comment (vmarkovtsev) [inline server/athenian/api/models/state/models.py:524]:
```suggestion
    If `time_from` and `time_to` are set then `time_interval` is NULL.
```

File: server/athenian/api/models/state/versions/7da0a85d2887_dashboards.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        ),
+        sa.Column("id", sa.Integer(), nullable=False),
+        sa.Column("dashboard_id", sa.Integer(), nullable=False),
+        sa.Column("position", sa.Integer(), nullable=False),
+        sa.Column("metric", sa.String(), nullable=False),
+        sa.Column("time_from", sa.TIMESTAMP(timezone=True), nullable=True),
+        sa.Column("time_to", sa.TIMESTAMP(timezone=True), nullable=True),
+        sa.Column("time_interval", sa.String(), nullable=True),
+        sa.Column("name", sa.String(), nullable=False),
+        sa.Column("description", sa.String(), nullable=False),
```
Comment (vmarkovtsev) [inline server/athenian/api/models/state/versions/7da0a85d2887_dashboards.py:72]: We can copy the comments here, too

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: server/athenian/api/models/state/versions/7da0a85d2887_dashboards.py (hunk 3 above)
Comment (gaetano-guerriero) [reply server/athenian/api/models/state/versions/7da0a85d2887_dashboards.py]: luckily autogeneration handled this

Comment (vmarkovtsev) [reply server/athenian/api/models/state/versions/7da0a85d2887_dashboards.py]: fancy!

**Review by vmarkovtsev — APPROVED** [review]

**Pull request merged**